api/client/annotations/client_test.go

```diff
@@ -4,90 +4,90 @@
 package annotations_test
 
 import (
+	"github.com/golang/mock/gomock"
 	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 
-	basetesting "github.com/juju/juju/api/base/testing"
+	basemocks "github.com/juju/juju/api/base/mocks"
 	"github.com/juju/juju/api/client/annotations"
 	"github.com/juju/juju/rpc/params"
-	coretesting "github.com/juju/juju/testing"
 )
 
 type annotationsMockSuite struct {
-	coretesting.BaseSuite
-	annotationsClient *annotations.Client
 }
 
 var _ = gc.Suite(&annotationsMockSuite{})
 
 func (s *annotationsMockSuite) TestSetEntitiesAnnotation(c *gc.C) {
-	var called bool
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
 	annts := map[string]string{"annotation1": "test"}
 	annts2 := map[string]string{"annotation2": "test"}
 	setParams := map[string]map[string]string{
 		"charmA":       annts,
 		"applicationB": annts2,
 	}
-	apiCaller := basetesting.APICallerFunc(
-		func(objType string,
-			version int,
-			id, request string,
-			a, result interface{},
-		) error {
-			called = true
-			c.Check(objType, gc.Equals, "Annotations")
-			c.Check(id, gc.Equals, "")
-			c.Check(request, gc.Equals, "Set")
 
-			args, ok := a.(params.AnnotationsSet)
-			c.Assert(ok, jc.IsTrue)
+	args := params.AnnotationsSet{
+		Annotations: []params.EntityAnnotations{
+			{
+				EntityTag:   "charmA",
+				Annotations: map[string]string{"annotation1": "test"},
+			},
+			{
+				EntityTag:   "applicationB",
+				Annotations: map[string]string{"annotation2": "test"},
+			},
+		},
+	}
+
+	for _, aParam := range args.Annotations {
+		// Since sometimes arrays returned on some
+		// architectures vary the order within params.AnnotationsSet,
+		// simply assert that each entity has its own annotations.
+		// Bug 1409141
+		c.Assert(aParam.Annotations, gc.DeepEquals, setParams[aParam.EntityTag])
+	}
```

Comment:
this code block should be done inside of the mock call by using DoAndReturn() instead of Return() on line 58.

example: https://github.com/juju/juju/blob/d70d0c27493529b70af9b7f379824f4b40b0d0cb/cmd/juju/application/deployer/bundlehandler_test.go#L1699